fileserver/cook/file_server.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+#
+#  Copyright (c) 2019 Two Sigma Open Source, LLC
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a copy
+#  of this software and associated documentation files (the "Software"), to
+#  deal in the Software without restriction, including without limitation the
+#  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+#  sell copies of the Software, and to permit persons to whom the Software is
+#  furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included in
+#  all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+#  IN THE SOFTWARE.
+#
+"""Module implementing a file server to serve Cook job logs. """
+
+import os
+from operator import itemgetter
+from pathlib import Path
+from stat import *
+
+import gunicorn.app.base
+from flask import Flask, jsonify, request, send_file
+from gunicorn.six import iteritems
+
+app = Flask(__name__)
+
+
+class FileServerApplication(gunicorn.app.base.BaseApplication):
+
+    def __init__(self, options=None):
+        self.options = options or {}
+        self.application = app
+        super(FileServerApplication, self).__init__()
+
+    def load_config(self):
+        config = dict([(key, value) for key, value in iteritems(self.options)
```

Comment:
```suggestion
        config = dict([(key, value) for key, value in self.options.items()
```

Reply:
Python 2.7+ supports _dict comprehensions_; i.e., no need to create lists of pairs and convert those to dicts:

```
        config = {key: value for key, value in self.options.items()
                  if key in self.cfg.settings and value is not None}
```

Reply:
Actually, can't we just move this `if` condition into the loop below and get rid of this whole intermediate collection?